src/Swift.Bindings/src/Emitter/StringCSharpEmitter/Handler/MethodHandler.cs

```diff
@@ -181,36 +122,48 @@ public class WrapperSignatureBuilder
     {
         private string _returnType = "invalid";
         private readonly List<Parameter> _parameters = new();
-        MethodDecl MethodDecl { get; }
-        BaseDecl ParentDecl { get; }
-        ITypeDatabase TypeDatabase { get; }
+        private readonly MethodEnvironment _env;
 
-        public WrapperSignatureBuilder(MethodDecl methodDecl, ITypeDatabase typeDatabase)
+        public WrapperSignatureBuilder(MethodEnvironment env)
         {
-            MethodDecl = methodDecl;
-            ParentDecl = methodDecl.ParentDecl!;
-            TypeDatabase = typeDatabase;
+            _env = env;
         }
 
         /// <summary>
         /// Handles the return type of the method.
         /// </summary>
         public void HandleReturnType()
         {
-            var argument = MethodDecl.CSSignature.First();
-            var typeRecord = TypeDatabase.GetTypeRecordOrAnyType(argument.SwiftTypeSpec);
-            SetReturnType(typeRecord.CSTypeIdentifier);
+            var argument = _env.MethodDecl.CSSignature.First();
+            if (argument.IsGeneric)
+            {
+                var CSName = _env.GenericTypeMapping[argument.SwiftTypeSpec.ToString()];
```

Comment:
Should `GenericTypeMapping` be part of the type database?

Reply:
I dont think so, this is just a per MethodEnvironment mapping, outside word should not see it.

Reply:
this are three things:
- valid placeholder name like `T`
- variable name for holding metadata for `T`
- variable name for holding payload of `T`

all of those are internal to the method